src/controllers/challenge.controllers.js

```diff
@@ -37,40 +37,47 @@ export const getChallengeById = async (req, res) => {
   }
 };
 
-// Add a member to a challenge
-export const addMember = async (req, res) => {
+// Remove member from challenge
+export const removeMember = async (req, res) => {
   try {
-    const { id } = req.params;
-    const { user_id } = req.body;
+    const { id, userId } = req.params;
     const challenge = await Challenge.findById(id);
     if (!challenge) {
       return res.status(404).json({ data: null, error: "Challenge not found" });
     }
-    challenge.members.push({ user_id });
+    const member = challenge.members.find(
+      (member) => member.user_id === userId,
+    );
+    if (!member) {
+      return res.status(404).json({ data: null, error: "Member not found" });
+    }
+    member.left_at = new Date();
     await challenge.save();
     res.status(200).json({ data: challenge, error: null });
   } catch (error) {
     res.status(500).json({ data: null, error: JSON.stringify(error) });
   }
 };
 
-// Remove member from challenge
-export const removeMember = async (req, res) => {
+// Join challenge
+export const joinChallenge = async (req, res) => {
   try {
-    const { id, userId } = req.params;
+    const { id } = req.params;
+    const { user_id, response } = req.body;
     const challenge = await Challenge.findById(id);
+
     if (!challenge) {
```

Comment:
* let's put a challenge status validation. Joiners shouldn't be able join closed challenges.
* handle duplicates (i.e. member will join again)

Reply:
Nice thanks for pointing that out. Addressed here: https://github.com/LangaraBeetles/weaup-api/pull/15/commits/7213865e1dabf4682d53d19c46c0fe67ea1f1fd8